core/src/main/java/com/amplifyframework/core/category/Category.java

```diff
@@ -221,6 +227,13 @@ protected P getSelectedPlugin() throws IllegalStateException {
                 "Tried to get a default plugin but there are more than one to choose from in this category. " +
                 "Call getPlugin(pluginKey) to choose the specific plugin you want to use in this case."
             );
+        } else if (State.FAILED.equals(state.get())) {
+            throw new IllegalStateException(
+                "Failed to get plugin because either configuration or initialization failed.  Check for " +
+                "configuration failures by logging any exceptions thrown by Amplify.configure().  Check for " +
+                "initialization failures by subscribing to the InitializationStatus.FAILED event on the Hub channel " +
+                "for the respective category with Amplify.Hub.subscribe()"
```

Comment:
I wonder if we should add a troubleshooting page in our docs, and link to it here?

Reply:
I kind of see this PR as a short term fix to be honest.  I think ideally, it would be better to just log the errors, inline, either when they occur, or here, when trying to get a plugin.

It is difficult to log the errors at the time that they occur, because the `Logging` plugin may not have been configured yet.   Maybe we could fix that by configuring it first though.

There is a `CategoryInitializationResult` object that contains the `Throwable` that caused an initialization failure, but we don't store this object currently.  Perhaps we could store it on `Category`, and then pass it back to the customer as the cause of the `IllegalStateException`?

@jamesonwilliams do you have any historical context on this on this `CategoryInitializationResult`?  Would it be reasonable to store it?

Reply:
Sure, here's a bit of background on that `CategoryInitializationResult`. The original assumption was that a category may have k-many plugins associated to it. (In practice: no, it's only ever one.)

Multiple plugins initialize in parallel, and the result of their initializations are captured into this `CategoryInitializationResult` container object.

The category as a whole is said to fail if any of the plugins failed to initialize. The container also provides some utility methods to inspect failures from the different plugins and see which plugins succeeded and failed.

In retrospect, it has turned out that 1 plugin implements 1 category spec, and we haven't ever needed to support multiple plugins per category. So this container object is overengineered, and we could simplify this part of the code considerably if we enforce a plugin count of 0 or 1 at a framework level.

Reply:
I just pushed a couple updates: 
 - Replaced `FAILED` with `CONFIGURATION_FAILED` and `INITIALIZATION_FAILED`, so that we can keep track of what failed, and provide a more specific error to the customer.
 - If the state is `INITIALIZATION_FAILED`, and the customer tries to get a plugin,  an `IllegalStateException` is thrown.  Now, that exception will also include the `Throwable` with the specific failure that occurred during initialization.  This is made possible by saving the `CategoryInitializationResult` on the `Category` object. 
 - Consolidated logic from the two very similar `getPlugin` methods into one helper method.